tests/Core/Ruleset/ExpandRulesetReferenceInternalTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+/**
+ * Test the Ruleset::expandRulesetReference() method.
+ *
+ * @author    Juliette Reinders Folmer <phpcs_nospam@adviesenzo.nl>
+ * @copyright 2025 PHPCSStandards and contributors
+ * @license   https://github.com/PHPCSStandards/PHP_CodeSniffer/blob/master/licence.txt BSD Licence
+ */
+
+namespace PHP_CodeSniffer\Tests\Core\Ruleset;
+
+use PHP_CodeSniffer\Ruleset;
+use PHP_CodeSniffer\Tests\ConfigDouble;
+use PHP_CodeSniffer\Tests\Core\Ruleset\AbstractRulesetTestCase;
+
+/**
+ * Test handling of "internal" references in the Ruleset::expandRulesetReference() method.
+ *
+ * @covers \PHP_CodeSniffer\Ruleset::expandRulesetReference
+ */
+final class ExpandRulesetReferenceInternalTest extends AbstractRulesetTestCase
+{
+
+
+    /**
+     * Verify that a ruleset reference starting with "Internal." (including the dot) doesn't cause any sniffs to be registered.
+     *
+     * @return void
+     */
+    public function testInternalRefDoesNotGetExpanded()
+    {
+        // Set up the ruleset.
+        $standard = __DIR__.'/ExpandRulesetReferenceInternalIgnoreTest.xml';
+        $config   = new ConfigDouble(["--standard=$standard"]);
+        $ruleset  = new Ruleset($config);
+
+        $expected = ['Generic.PHP.BacktickOperator' => 'PHP_CodeSniffer\\Standards\\Generic\\Sniffs\\PHP\\BacktickOperatorSniff'];
+
+        $this->assertSame($expected, $ruleset->sniffCodes);
+
+    }//end testInternalRefDoesNotGetExpanded()
```

Comment:
Should we have anything here to ensure that the `<severity>` that's set for an internal code is read correctly?

Reply:
Good point, but that should be tested via another method/test class. I think it would be good to add an additional test to the `ProcessRuleShouldProcessElementTest` class for that.

Reply:
See PR #818